src/services/outliningElementsCollector.ts

```diff
@@ -161,6 +173,10 @@ namespace ts.OutliningElementsCollector {
                 case SyntaxKind.CaseBlock: {
                     const openBrace = findChildOfKind(n, SyntaxKind.OpenBraceToken, sourceFile);
                     const closeBrace = findChildOfKind(n, SyntaxKind.CloseBraceToken, sourceFile);
+                    if (n.kind === SyntaxKind.ObjectLiteralExpression && n.parent.kind === SyntaxKind.ArrayLiteralExpression) {

```

Comment:
I'm afraid I don't understand why this case is special.  It looks like the main things we're doing differently are calling `getStart` instead of `pos` and computing our own `hintSpan`.  Why not make those changes to the general case?  If this case is indeed different from the general case, why aren't there related cases (nested arrays, for example)?